Hulya form modal

hulyak opened the pull request:
Created the form modal and confirmation component. We have some duplicate code for `Get in Touch` text coming from the Review component. We need to remove that.  Form doesn't have validation yet. 
![Screen Shot 2021-09-30 at 11 50 34 AM](https://user-images.githubusercontent.com/57311842/135513729-0fa8e5c1-d298-4ea8-a501-5a713f08653b.png)
![Screen Shot 2021-09-30 at 11 50 03 AM](https://user-images.githubusercontent.com/57311842/135513725-cb75f279-2ef0-4f5c-a06d-5ffeeb55157a.png)


![Screen Shot 2021-09-30 at 11 49 34 AM](https://user-images.githubusercontent.com/57311842/135513722-3ef5a3ae-426c-4751-8424-aaf4914190cb.png)

hulyak committed:
create example modal

hulyak committed:
create the form styles

hulyak committed:
fix merge conflict

hulyak committed:
add imports

hulyak committed:
edit input

hulyak committed:
create state object for holding input fields

hulyak committed:
edit styles

hulyak committed:
close popup with ESC key and outside click

hulyak committed:
add react spring

hulyak committed:
add new components

hulyak committed:
update modal and confirmation box

hulyak committed:
change handler name

hulyak committed:
update styles

hulyak committed:
update the structure of components

hulyak committed:
update modal and confirmation

hulyak committed:
create different component for Modal info

hulyak committed:
edit form

hulyak committed:
remove react spring

hulyak committed:
update code

hulyak committed:
Merge branch 'main' of https://github.com/Muddoo/All_Correct into hk-form

hulyak committed:
update the modal

hulyak committed:
remove errors

sayus2884 reviewed (changes requested):
Hey @hulyak. 

Here are the things that needs minor fixing:

- Rename the `Form` to something specific like `GetInTouch`. `Form` can be perceived as a reusable component by another developer when it's actually a section of the main page.
- Rename the `Modal` to something specific like `GetInTouchModal` since it's coded specifically for the get in touch form. Same reason as above.
- Place ErrorBoundary.js inside ErrorBoundary folder.
- The form should also be accessed by the navbar.

![navbar can do it too](https://user-images.githubusercontent.com/5374194/135693329-438b3058-e613-42ba-9ee9-59f475847d63.PNG)

- Was unable to close the modal with mouse click. `Background` was nowhere to be found (no blue highlight).

![no background](https://user-images.githubusercontent.com/5374194/135694212-bd6591ac-4159-4c92-8f72-546f431254ee.png)

**Optional**
- There may be cases where the height of the modal is longer compared to the device height resulting in hidden inputs like the image above. It may be good to make the modal scrollable by setting it's container `height:100vh` and `overflow-y: auto`. Reference [here](https://stackoverflow.com/questions/1575141/how-to-make-a-div-100-height-of-the-browser-window).

- I recommend setting a dark transparent overlay and [removing the scrollable body](https://stackoverflow.com/questions/54989513/react-prevent-scroll-when-modal-is-open) so the user's focus is on the form and not on anything else. Not essential, quality-of-life feature.

- Reset the form when the modal is closed. This is so the user can send another one if they messed up the first time around.

Also, I don't see any height problem that you mentioned on slack, It's working fine on my Chrome and Edge. How do you reproduce the problem?

Other than above, code's looking solid and is easy to follow.

hulyak committed:
update component names and modal height

hulyak committed:
lift the state up for toggling the modal for Navigation and GetInTouch Form

hulyak committed:
pass modal props to footer and footer navigation

hulyak commented:
> Hey @hulyak.
> 
> Here are the things that needs minor fixing:
> 
> * Rename the `Form` to something specific like `GetInTouch`. `Form` can be perceived as a reusable component by another developer when it's actually a section of the main page.
> * Rename the `Modal` to something specific like `GetInTouchModal` since it's coded specifically for the get in touch form. Same reason as above.
> * Place ErrorBoundary.js inside ErrorBoundary folder.
> * The form should also be accessed by the navbar.
> 
> ![navbar can do it too](https://user-images.githubusercontent.com/5374194/135693329-438b3058-e613-42ba-9ee9-59f475847d63.PNG)
> 
> * Was unable to close the modal with mouse click. `Background` was nowhere to be found (no blue highlight).
> 
> ![no background](https://user-images.githubusercontent.com/5374194/135694212-bd6591ac-4159-4c92-8f72-546f431254ee.png)
> 
> **Optional**
> 
> * There may be cases where the height of the modal is longer compared to the device height resulting in hidden inputs like the image above. It may be good to make the modal scrollable by setting it's container `height:100vh` and `overflow-y: auto`. Reference [here](https://stackoverflow.com/questions/1575141/how-to-make-a-div-100-height-of-the-browser-window).
> * I recommend setting a dark transparent overlay and [removing the scrollable body](https://stackoverflow.com/questions/54989513/react-prevent-scroll-when-modal-is-open) so the user's focus is on the form and not on anything else. Not essential, quality-of-life feature.
> * Reset the form when the modal is closed. This is so the user can send another one if they messed up the first time around.
> 
> Also, I don't see any height problem that you mentioned on slack, It's working fine on my Chrome and Edge. How do you reproduce the problem?
> 
> Other than above, code's looking solid and is easy to follow.

Thanks for the review @sayus2884. I have changed the names of the components and made the modal available in the navigation of the header and footer. For the height, I have added scroll for small screen sizes, and reset the form fields whenever the user closes the form.

hulyak committed:
instead of prop drilling, created a custom hook for GetInTouchModal and used Context for passing the props

hulyak committed:
update the modal with context
